kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSValueParameterImpl.kt

```diff
@@ -77,11 +81,26 @@ class KSValueParameterImpl private constructor(
     override val isCrossInline: Boolean
         get() = ktValueParameterSymbol.isCrossinline
 
+    private val KaValueParameterSymbol.primaryConstructorProperty: KaPropertySymbol? by lazy {
+        when (ktValueParameterSymbol.origin) {
+            // ktValueParameterSymbol.generatedPrimaryConstructorProperty is always null in libraries.
+            // TODO: fix in AA
+            KaSymbolOrigin.LIBRARY, KaSymbolOrigin.JAVA_LIBRARY -> analyze {
+                val cstr = ktValueParameterSymbol.containingDeclaration as? KaConstructorSymbol
+                val cls = cstr?.containingDeclaration as? KaClassSymbol
+                cls?.declaredMemberScope?.declarations?.filterIsInstance<KaPropertySymbol>()
+                    ?.firstOrNull { it.name == ktValueParameterSymbol.name }
```

Comment:
In addition to `name` check, you can use `isFromPrimaryConstructor`: https://github.com/JetBrains/kotlin/blob/master/analysis/analysis-api/src/org/jetbrains/kotlin/analysis/api/symbols/KaVariableSymbol.kt#L245 ?

Reply:
Unfortunately `isFromPrimaryConstructor` is also PSI based and doesn't work on libraries.

But that reminds me of a case that this patch doesn't cover:
```
class MyClass(prop: Int = 1) {
    val prop: Int = 0
}
```

IMHO it is impossible to distinguish without adding metadata in the libraries.

Anyway, this patch covers most of the use cases and doesn't introduce new issues. If there are no other concerns, I'm going to merge it.